adagrams/game.py

```diff
@@ -1,11 +1,144 @@
+from multiprocessing import pool
+import random
+
+def build_letter_pool():
+    LETTER_POOL = {
+            "A": 9,
+            "B": 2,
+            "C": 2,
+            "D": 4,
+            "E": 12,
+            "F": 2,
+            "G": 3,
+            "H": 2,
+            "I": 9,
+            "J": 1,
+            "K": 1,
+            "L": 4,
+            "M": 2,
+            "N": 6,
+            "O": 8,
+            "P": 2,
+            "Q": 1,
+            "R": 6,
+            "S": 4,
+            "T": 6,
+            "U": 4,
+            "V": 2,
+            "W": 2,
+            "X": 1,
+            "Y": 2,
+            "Z": 1
+            }
+
+    return LETTER_POOL
+
+
+def build_available_letters_list(LETTER_POOL):
+    available_letters = []
+
+    for letter,frequency in LETTER_POOL.items():
+        for i in range(frequency):
+            available_letters.append(letter)
+
+    return available_letters
+
+
 def draw_letters():
-    pass
+    hand = []
+    LETTER_POOL = build_letter_pool()
+    available_letters = build_available_letters_list(LETTER_POOL)
+    count_dict = {}
+    
+    while len(hand) < 10:
+        random_letter = random.choice(available_letters)
+        
+        if random_letter not in hand:
+            hand.append(random_letter)
+            count_dict[random_letter] = 1
+        elif LETTER_POOL[random_letter] > count_dict[random_letter]:
+            hand.append(random_letter)
+            count_dict[random_letter] += 1
+
+    return hand
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letter_bank_copy = letter_bank.copy()
+    for letter in word.upper():
+        if letter not in letter_bank_copy:
+            return False
+        
+        letter_bank_copy.remove(letter)
+    
+    return True
+
+
+def build_score_chart():
+    """
+    |Letter                        | Value|
+    |:----------------------------:|:----:|
+    |A, E, I, O, U, L, N, R, S, T  |   1  |
+    |D, G                          |   2  |
+    |B, C, M, P                    |   3  |
+    |F, H, V, W, Y                 |   4  |
+    |K                             |   5  |
+    |J, X                          |   8  |
+    |Q, Z                          |   10 |
+    """
+
+    score_chart = {
```

Comment:
Since this helper just returns a constant variable, you can make score_chart a global constant called SCORE_CHART that is declared at the top of this file.